packages/xo-web/src/common/xo/index.js

```diff
@@ -958,11 +958,14 @@ export const copyVm = (vm, sr, name, compress) => {
         title: _('copyVm'),
         body: <CopyVmModalBody vm={vm} />,
       }).then(params => {
-        if (!params.sr) {
+        const fullCopy = params.copyMode === 'fullCopy'
+        if (fullCopy && !params.sr) {
           error('copyVmsNoTargetSr', 'copyVmsNoTargetSrMessage')
           return
         }
-        return _call('vm.copy', { vm: vmId, ...params })
+        return fullCopy
+          ? _call('vm.copy', { vm: vmId, ...params })
+          : cloneVm({ id: vmId, name_label: params.name })
```

Comment:
This is weird. There's a high-level call with a low-level call. Maybe we should have a simple `copyVm` without modal prompts like `cloneVm` and then handle all the modal stuff in the high-level function. What do you think?